### DISCUSSION TITLE
Use of EFS for Data Persistence on Milvus

### DISCUSSION THREAD

**Discussion by ganderaj** [milvus-io/milvus#26527] [Q&A and General discussion]
We have deployed Milvus (v2.2.13) on AWS EKS which went into unhealthy state post applying HPA rules (not sure if it caused it though).

Pod Error Messages as per container logs --

On Rootcoord
[error="find no available querycoord, check querycoord state"]
[error="err: find no available datacoord, check datacoord state"]

On Datacord and Querycord
[error="find no available rootcoord, check rootcoord state"]

Environment was back to being stable 20m later after undergoing couple of pod restarts across several components. However, **ALL DATA WAS LOST**. The databases, collections, roles, grants, users among others were all purged. It appeared as if it is a fresh new installation.

Questions --

- Where is the data concering Databases, Collections etc., is stored ?
- Assuming PVCs backing ETCD hold such data, is it expected of Milvus to lose data once the underlying EKS Node is scaled down ?
- Can we use EFS instead of EBS to support PV / PVC for ETCD ?

**Comment by xiaofan-luan** [comment]
I just replied on the other other issue.
My guess is that your etcd data lost for some reason, and it's highly likely an ETCD deployment issue or your Milvus configure the wrong etcd path

**Reply by ganderaj** [reply]
As a part of applying HPAs, I have update my deployments with "resources" per Milvus sizing tool which also provided few ENV variables for etcd. Variables included below were modified / added but nothing concerning root directory.

  - name: ETCD_QUOTA_BACKEND_BYTES
    value: "4294967296"
  - name: ETCD_HEARTBEAT_INTERVAL
    value: "500"
  - name: ETCD_ELECTION_TIMEOUT
    value: "25000"
  - name: ETCD_SNAPSHOT_COUNT
    value: "10000"
  - name: ETCD_ENABLE_PPROF
    value: "true"

**Comment by xiaofan-luan** [comment]
And EFS performance is actaully not as good as EBS for etcd storage but it's actually doable for case without frequent meta update
